src/View/Helper/DocBlockHelper.php

```diff
@@ -51,23 +89,49 @@ public function buildEntityPropertyHintTypeMap(array $propertySchema)
     {
         $properties = [];
         foreach ($propertySchema as $property => $info) {
-            switch ($info['kind']) {
-                case 'column':
-                    $properties[$property] = $this->columnTypeToHintType($info['type']);
-                    break;
-
-                case 'association':
-                    $type = $this->associatedEntityTypeToHintType($info['type'], $info['association']);
-                    if ($info['association']->type() === Association::MANY_TO_ONE) {
-                        $properties = $this->_insertAfter(
-                            $properties,
-                            $info['association']->foreignKey(),
-                            [$property => $type]
-                        );
-                    } else {
-                        $properties[$property] = $type;
-                    }
-                    break;
+            if ($info['kind'] == 'column') {
+                $properties[$property] = $this->columnTypeToHintType($info['type']);
+            }
+        }
+        return $properties;
+    }
+
+    /**
+     * Builds a map of entity associations as DocBlock types for use
+     * in generating `@property` hints.
+     *
+     * This method expects a property schema as generated by
+     * `\Bake\Shell\Task\ModelTask::getEntityPropertySchema()`.
+     *
+     * The generated map has the format of
+     *
+     * ```
+     * [
+     *     'property-name' => 'doc-block-type',
+     *     ...
+     * ]
+     * ```
+     *
+     * @see \Bake\Shell\Task\ModelTask::getEntityPropertySchema
+     *
+     * @param array $propertySchema The property schema to use for generating the type map.
+     * @return array The property DocType map.
+     */
+    public function buildEntityAssociationHintTypeMap(array $propertySchema)
+    {
+        $properties = [];
+        foreach ($propertySchema as $property => $info) {
+            if ($info['kind'] == 'association') {
+                $type = $this->associatedEntityTypeToHintType($info['type'], $info['association']);
+                if ($info['association']->type() === Association::MANY_TO_ONE) {
+                    $properties = $this->_insertAfter(
```

Comment:
If you split the properties, then doing this isn't necessary anymore, as it is used to position the association properties after the corresponding foreign key properties (`kind = column`).